File: Benchmarks_MCS-IIS/Programs_Benchmarks_MCS-IIS_BugAssist/0_Conditions_Fausses/ArrayInitKO.c


 
   
   
   
   
void ArrayInitKO(){
  int i;int a[10];
  i=0;
  while(i<10){
    a[i]=0;  // error in the assignment : should be "a[i]=i+1"
    i=i+1;
  }
  assert(a[8] != 0);
}

File: Benchmarks_MCS-IIS/Programs_Benchmarks_MCS-IIS/0_Conditions_Fausses/ArrayInitKO.java
class array_int{

 /*@ requires 
   @(\forall int k;(k >= 0 && k < a.length);a[k] == 0);
   @ ensures 
   @ (a[8] != 0); 
   @*/
int ArrayInitKO(int[] a){
  int i;
  i=0;
  while(i<a.length){
    a[i]=0;  // error in the assignment : should be "a[i]=i+1"
    i=i+1;
  }
  return 0;
}
}
